### ISSUE TITLE
Doubly-encoded UTF-8 instead of using substitution of invalid bytes

### ISSUE THREAD

**Issue by dato** [djcb/mu#949]
TL;DR: mu double-encodes the [attached mail](https://github.com/djcb/mu/files/596156/example.txt) instead of trusting the header and substituting the invalid 0xE1 byte with a `?` character.

```
% mu --version
mu (mail indexer/searcher) version 0.9.16
```
-----

Hi!

I have the following situation: somebody sends to my (University) address a mail in _text/plain; charset=utf-8_. Prior to delivering to my mailbox, the Uni’s servers insert a footer which contains a non-ASCII character, and it inserts it as a single byte 0xE1 (which is Latin1 for `á`).

This is _very_ broken indeed, and I’ve already reported to my admins. But, in the meantime, I would need a workaround.

I can instruct editors like Vim and e-mail clients like Mutt to _trust_ the Content-Type header and treat the body as UTF-8—rendering the 0xE1 byte as `?` because, of course, it’s not valid UTF-8.

Now if I do `mu view ./example` (see attached), this is what I see in my UTF-8 terminal:

```
From: Friend <friend@example.org>
To: Me <me@example.org>
Subject: test double UTF-8
Date: Wed Nov 16 20:17:39 2016
Somebody writes to me: Ã¡h!

-- 
This footer gets inserted: báh.
```

whereas I would like for mu to output UTF-8 without doubly-converting it:

```
From: Friend <friend@example.org>
To: Me <me@example.org>
Subject: test double UTF-8
Date: Wed Nov 16 20:17:39 2016
Somebody writes to me: áh!

-- 
This footer gets inserted: b?h.
```

But mu’s output is double-utf8-encoded, and the latter only happens if I do: `mu view ./example | recode utf-8..latin1`.

**Comment by djcb** [comment]
I'm not planning to add things for broken email... there are just too many cases, and there will be exceptions etc... so a bit too much added complexity for a corner-case.

Now, that doesn't solve your problem of course -- I think the easiest solution for that is to use something like
```
sed 's/=E1/=C3=A1/g' < example.txt > example2.txt
```
to fix your email, perhaps as an extra step after getting mail.

**Comment by dato** [comment]
Respectfully, I'm not asking for a workaround. I'm asking for a
fix of mu's broken behavior.

Mu is parsing a MIME part marked as UTF-8 and singlehandedly
deciding that, since one character or two were invalid UTF-8, that
means the whole thing should be read as Latin1, and back-encoded
to UTF-8 on output.

That, to me, reads as "let's allow broken Content-Type headers".

The less intrusive approach would be, of course, to use
substitution characters for the invalid sequences.

-d

**Comment by dato** [comment]
Additionally, if you would prefer not to honor a broken
Content-Type header (which is, after all, reasonable), the
fall-back should not be Latin1 inconditionally, but the system's
locale, whichever it might be.

I hope you can be persuaded by this argument at least.

Thanks,

-d
